packaging/flavorgen/flavors/generators.go

```diff
@@ -0,0 +1,376 @@
+	kubeadmv1beta1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+)
+
+const (
+	clusterNameVar              = "${ CLUSTER_NAME }"
+	namespaceVar                = "${ NAMESPACE }"
+	defaultNumCPUs              = 2
+	defaultCloudProviderImage   = "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.0.0"
+	kubernetesVersionVar        = "${ KUBERNETES_VERSION }"
+	workerMachineCountVar       = "${ WORKER_MACHINE_COUNT }"
+	controlPlaneMachineCountVar = "${ CONTROL_PLANE_MACHINE_COUNT }"
+	folderVar                   = "${ VSPHERE_FOLDER_VAR }"
+	machineDeploymentNameSuffix = "-md-0"
+	resourcePoolVar             = "${ VSPHERE_RESOURCE_POOL }"
+	sshAuthorizedKeysVar        = "${ SSH_AUTHORIZED_KEY }"
+	vSphereDataCenterVar        = "${ VSPHERE_DATACENTER }"
+	vSphereDatastoreVar         = "${ VSPHERE_DATASTORE }"
+	vSphereNetworkVar           = "${ VSPHERE_NETWORK }"
+	vSphereServerVar            = "${ VSPHERE_SERVER }"
+	vSphereTemplateVar          = "${ VSPHERE_TEMPLATE }"
+	defaultHAProxyTemplate      = "capv-haproxy-v0.5.3-77-g224e0ef6"
+	defaultClusterCIDR          = "192.168.0.0/16"
+	defaultDiskGiB              = 25
+	defaultMemoryMiB            = 8192
+)
+
+type replacement struct {
+	kind      string
+	name      string
+	value     interface{}
+	fieldPath []string
+}
+
+var replacements = []replacement{
+	{
+		kind:      "KubeadmControlPlane",
+		name:      "${ CLUSTER_NAME }",
+		value:     controlPlaneMachineCountVar,
+		fieldPath: []string{"spec", "replicas"},
+	},
+	{
+		kind:      "MachineDeployment",
+		name:      "${ CLUSTER_NAME }-md-0",
+		value:     workerMachineCountVar,
+		fieldPath: []string{"spec", "replicas"},
+	},
+}
+
+func newVSphereCluster(lb *infrav1.HAProxyLoadBalancer) infrav1.VSphereCluster {
+	vsphereCluster := infrav1.VSphereCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: infrav1.GroupVersion.String(),
+			Kind:       typeToKind(&infrav1.VSphereCluster{}),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterNameVar,
+			Namespace: namespaceVar,
+		},
+		Spec: infrav1.VSphereClusterSpec{
+			Server: vSphereServerVar,
+			CloudProviderConfiguration: infrav1.CPIConfig{
+				Global: infrav1.CPIGlobalConfig{
+					SecretName:      "cloud-provider-vsphere-credentials",
+					SecretNamespace: metav1.NamespaceSystem,
+					Insecure:        true,
+				},
+				VCenter: map[string]infrav1.CPIVCenterConfig{
+					vSphereServerVar: {Datacenters: vSphereDataCenterVar},
+				},
+				Network: infrav1.CPINetworkConfig{
+					Name: vSphereNetworkVar,
+				},
+				Workspace: infrav1.CPIWorkspaceConfig{
+					Server:       vSphereServerVar,
+					Datacenter:   vSphereDataCenterVar,
+					Datastore:    vSphereDatastoreVar,
+					ResourcePool: resourcePoolVar,
+					Folder:       folderVar,
+				},
+				ProviderConfig: infrav1.CPIProviderConfig{
+					Cloud: &infrav1.CPICloudConfig{
+						ControllerImage: defaultCloudProviderImage,
```

Comment:
shouldn't be this accessed through `cloudprovidersvc` too ? (i.e. we should add it if not)